### PR TITLE
My hw 04

### PR DESCRIPTION
**Pull request by manzhaimaksim**

### CODE REVIEW

File: HW04/maksim_manzhai/lib/homework.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+class Homework
+  attr_reader :title
+  attr_accessor :mentor, :answers, :student, :acceptable
+
+  def initialize(attributes)
+    @title = attributes.fetch(:title)
+    @description = attributes.fetch(:description)
+    @student = attributes.fetch(:student)
+    @mentor = attributes.fetch(:mentor)
+    @student.notifications << Notification.new(title: @title, description: @description)
```
Comment (B0RlS) [inline HW04/maksim_manzhai/lib/homework.rb:12]: don't create anything in the initialize method

File: HW04/maksim_manzhai/lib/homework.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class Homework
+  attr_reader :title
+  attr_accessor :mentor, :answers, :student, :acceptable
+
+  def initialize(attributes)
```
Comment (B0RlS) [inline HW04/maksim_manzhai/lib/homework.rb:7]:
```suggestion
  def initialize(title:, description:, student:, mentor:)
```

File: HW04/maksim_manzhai/lib/homework.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+class Homework
+  attr_reader :title
+  attr_accessor :mentor, :answers, :student, :acceptable
+
+  def initialize(title:, description:, student:, mentor:)
+    @title = title
+    @description = description
+    @student = student
+    @mentor = mentor
+    send_notification
```
Comment (B0RlS) [inline HW04/maksim_manzhai/lib/homework.rb:12]:
```suggestion
```

File: HW04/maksim_manzhai/lib/homework.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    @answers = []
+    @acceptable = false
+  end
+
+  def send_notification
+    @student.notifications << Notification.new(title: @title, description: @description)
+  end
```
Comment (B0RlS) [inline HW04/maksim_manzhai/lib/homework.rb:19]:
```suggestion
  def send_notification_to_student
    student.notifications << Notification.new(title: title, description: description)
  end
```

File: HW04/maksim_manzhai/lib/homework.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Homework
+  attr_reader :title
+  attr_accessor :mentor, :answers, :student, :acceptable
```
Comment (B0RlS) [inline HW04/maksim_manzhai/lib/homework.rb:5]:
```suggestion
  attr_reader :title, :mentor, : student
  attr_accessor :answers, :acceptable
```

File: HW04/maksim_manzhai/lib/mentor.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    @notifications = []
+  end
+
+  def add_homework(attributes)
+    @title = attributes.fetch(:title)
+    @description = attributes.fetch(:description)
+    @student = attributes.fetch(:student)
+    @mentor = attributes.fetch(:mentor)
+    Homework.new(title: @title, description: @description, student: @student, mentor: @mentor)
+  end
```
Comment (B0RlS) [inline HW04/maksim_manzhai/lib/mentor.rb:19]:
```suggestion
  def add_homework(title:, description:, student:)
    Homework.new(title: title, description: description, student: student, mentor: <call yourself here>)
  end
```

File: HW04/maksim_manzhai/lib/mentor.rb [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class Mentor
+  attr_accessor :subscriptions, :notifications
```
Comment (B0RlS) [inline HW04/maksim_manzhai/lib/mentor.rb:4]:
```suggestion
  attr_reader :name, :surname, :subscriptions, :notifications
```

File: HW04/maksim_manzhai/lib/mentor.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    @subscriptions = []
+    @notifications = []
+  end
+
+  def add_homework(title:, description:, student:)
+    Homework.new(title: title, description: description, student: student, mentor: self)
+  end
+
+  def subscribe_to!(student)
+    @subscriptions << student
```
Comment (DashaSledopit) [inline HW04/maksim_manzhai/lib/mentor.rb:18]:
```suggestion
    subscriptions << student
```

File: HW04/maksim_manzhai/lib/mentor.rb [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  def mark_as_read!
+    @notifications.each { |notification| notification.readed = true }
```
Comment (DashaSledopit) [inline HW04/maksim_manzhai/lib/mentor.rb:22]:
```suggestion
    notifications.each { |notification| notification.readed = true }
```

File: HW04/maksim_manzhai/lib/student.rb [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+  def initialize(name:, surname:)
+    @name = name
+    @surname = surname
+    @notifications = []
+    @homeworks = []
+  end
+
+  def mark_as_read!
+    @notifications.each { |notification| notification.readed = true }
```
Comment (DashaSledopit) [inline HW04/maksim_manzhai/lib/student.rb:14]:
```suggestion
    notifications.each { |notification| notification.readed = true }
```

File: HW04/maksim_manzhai/lib/student.rb [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def to_work!(homework)
+    @homeworks << homework
```
Comment (DashaSledopit) [inline HW04/maksim_manzhai/lib/student.rb:18]:
```suggestion
    homeworks << homework
```

File: HW04/maksim_manzhai/lib/homework.rb [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    @title = title
+    @description = description
+    @student = student
+    @mentor = mentor
+    send_notification_to_student
+    @answers = []
+    @acceptable = false
+  end
+
+  def send_notification_to_student
```
Comment (DashaSledopit) [inline HW04/maksim_manzhai/lib/homework.rb:17]: Why do you need this method?

Comment (manzhaimaksim) [reply HW04/maksim_manzhai/lib/homework.rb:17]: we need send notification to student about new homework

Comment (DashaSledopit) [reply HW04/maksim_manzhai/lib/homework.rb:17]: so, you can do it in your `add_homework` method :)

**Pull request merged**